doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+# Metrics
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/metrics.js -->
+
+The `node:metrics` module provides an API for application instrumentation and
+performance monitoring. It offers various metric types and built-in exporters
+for popular monitoring systems.
+
+The module can be accessed using:
+
+```mjs
+import * as metrics from 'node:metrics';
+```
+
+```cjs
+const metrics = require('node:metrics');
+```
+
+## Overview
+
+The metrics API enables developers to instrument their applications with custom
+metrics that can be collected and exported to monitoring systems. All metrics
+publish their data through the `node:diagnostics_channel` module, allowing for
+flexible consumption patterns.
+
+### Example
+
+```mjs
+import { counter, timer, statsdStream } from 'node:metrics';
+import { createWriteStream } from 'node:fs';
+
+// Create a counter metric
+const apiCalls = counter('api.calls', { service: 'web' });
+
+// Create a timer factory
+const requestTimer = timer('api.request.duration', { service: 'web' });
+
+// Export metrics to StatsD format
+const statsd = statsdStream();
+statsd.pipe(createWriteStream('metrics.log'));
+
+// Use metrics in your application
+function handleRequest(req, res) {
+  const timer = requestTimer.create({ endpoint: req.url });
+
+  apiCalls.increment();
+
+  // Process request...
+
+  timer.stop();
+}
+```
+
+```cjs
+const { counter, timer, statsdStream } = require('node:metrics');
+const { createWriteStream } = require('node:fs');
+
+// Create a counter metric
+const apiCalls = counter('api.calls', { service: 'web' });
+
+// Create a timer factory
+const requestTimer = timer('api.request.duration', { service: 'web' });
+
+// Export metrics to StatsD format
+const statsd = statsdStream();
+statsd.pipe(createWriteStream('metrics.log'));
+
+// Use metrics in your application
+function handleRequest(req, res) {
+  const timer = requestTimer.create({ endpoint: req.url });
+
+  apiCalls.increment();
+
+  // Process request...
+
+  timer.stop();
+}
+```
+
+## Metric Types
+
+### `metrics.counter(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the counter metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Counter}
+
+Creates a counter metric that tracks cumulative values.
+
+```mjs
+import { counter } from 'node:metrics';
+
+const errorCount = counter('errors.total', { component: 'database' });
+
+errorCount.increment();     // Increment by 1
+errorCount.increment(5);    // Increment by 5
+errorCount.decrement(2);    // Decrement by 2
+```
```

Comment:
For most of these, it would also be helpful for the metric to provide a counter for the number of discreet observations made. E.g. for `counter`, it would be the number of times `increment`/`decrement` are called, etc. For an example of how this could be used, on `uniqueSet`, it can be used to calculate the percentage of observations that end up being unique or duplicated